integrated_channels/integrated_channel/course_metadata.py

```diff
@@ -0,0 +1,127 @@
+"""
+Assist integrated channels with retrieving course metadata.
+
+Module contains resources for integrated pipelines to retrieve all the
+metadata for courses in the course catalog belonging to a particular
+enterprise customer.
+"""
+from __future__ import absolute_import, unicode_literals
+
+import json
+from logging import getLogger
+
+from django.contrib.auth.models import User
+from slumber.exceptions import SlumberHttpBaseException
+
+from enterprise.course_catalog_api import CourseCatalogApiClient
+from enterprise.models import EnterpriseCustomer
+
+
+EXCLUDED_COURSE_DETAIL_KEYS = [
+    'course_runs',
+]
+
+
+LOGGER = getLogger(__name__)
+
+
+def updated_course_run(course_details, course_run_details):
+    """
+    Update a course run-specific details dict with details from the base course.
+
+    Args:
+        course_details (dict): The details of the base course
+        course_run_details (dict): The details of the specific course run
+    """
+    course_details = course_details.copy()
+    course_run_details = course_run_details.copy()
+    for key, base_course_value in course_details.items():
+        if course_run_details.get(key) is None and key not in EXCLUDED_COURSE_DETAIL_KEYS:
+            course_run_details[key] = base_course_value
+    return course_run_details
+
+
+def get_course_list(user, enterprise_customer):
+    """
+    Determine which catalog belongs to a given Enterprise Customer and list the course runs it contains.
+
+    Args:
+        user: A Django user requesting the course list
+        enterprise_customer: The given Enterprise Customer
+
+    Returns:
+        iterable: An iterable containing the details of each course run.
+    """
+    catalog_id = enterprise_customer.catalog
+
+    if catalog_id is None:
+        LOGGER.info('No catalog attached to %s', enterprise_customer)
```

Comment:
I think it's best to filter out any `enterprise_customer` objects with no `catalog` way back in the management command, and just raise here.

Or better yet, take a `catalog_id` parameter, instead of `enterprise_customer`.